File: avalon-support/avalon-security-support/src/main/java/io/github/lishangbu/avalon/security/properties/SecurityProperties.java
package io.github.lishangbu.avalon.security.properties;

import java.util.ArrayList;
import java.util.List;
import org.springframework.boot.context.properties.ConfigurationProperties;

/**
 * 安全配置项
 *
 * @author lishangbu
 * @since 2023/10/6
 */
@ConfigurationProperties(prefix = SecurityProperties.SECURITY_PROPERTIES_PREFIX)
public class SecurityProperties {

  public static final String SECURITY_PROPERTIES_PREFIX = "spring.security";

  private List<String> ignoreUrls = new ArrayList<>();

  public List<String> getIgnoreUrls() {
    return ignoreUrls;
  }

  public void setIgnoreUrls(List<String> ignoreUrls) {
    this.ignoreUrls = ignoreUrls;
  }
}

File: security/src/main/kotlin/io/github/lishangbu/security/config/SecurityProperties.kt
package io.github.lishangbu.security.config

import org.springframework.boot.context.properties.ConfigurationProperties

/**
 * 安全模块运行时配置。
 *
 * OAuth client、token 策略和密钥等持久化资源由数据库管理；配置文件只保留启停和
 * issuer 这类应用装配参数。
 */
@ConfigurationProperties("backend.security")
data class SecurityProperties(
	val enabled: Boolean = true,
	val issuer: String = "http://localhost:8080",
)
